tools/validate/validate.cpp

```diff
@@ -100,10 +105,17 @@ void scan_file(const filesystem::path& filepath, const TabPolicy tab_policy, vec
                 }
             }
 
-            if ((prev == ' ' || prev == '\t') && (ch == CR || ch == LF)) {
-                ++trailing_whitespace_lines;
-            }
+            if (ch == CR || ch == LF) {
+                if (prev == ' ' || prev == '\t') {
+                    ++trailing_whitespace_lines;
+                }
 
+                if (columns > max_line_length) {
+                    ++overlength_lines;
+                }
+                columns = ~size_t{0};
+            }
+            ++columns;

```

Comment:
While correct, I think this logic is *slightly* confusing. It would seem to be simpler to say:
```suggestion
                columns = 0;
            } else {
                ++columns;
            }
```
This avoids having to reason about the "fall through" modification, and makes it clear what the possible modifications to `columns` are - we either reset it to `0` or increment it.